MSAL/test/unit/native_auth/public/MSALNativeAuthPublicClientApplicationTest.swift

```diff
@@ -364,4 +377,505 @@ final class MSALNativeAuthPublicClientApplicationTest: XCTestCase {
+                                                                         tokenResponseValidator: tokenResponseValidatorMock)
+        let signUpController = MSALNativeAuthSignUpController(config: configuration,
+                                                              requestProvider: signUpRequestProviderMock,
+                                                              responseValidator: signUpResponseValidatorMock,
+                                                              signInController: signInAfterSignUpController)
+        
+        let controllerFactory = MSALNativeAuthControllerProtocolFactoryMock(signUpController: signUpController)
+        
+        sut = MSALNativeAuthPublicClientApplication(
+            controllerFactory: controllerFactory,
+            inputValidator: MSALNativeAuthInputValidator(),
+            internalChallengeTypes: []
+        )
+        
+        // Correlation Id is validated internally against expectedStartRequestParameters and expectedChallengeRequestParameters
+        sut.signUp(username: "username", attributes: ["key": "value"], correlationId: correlationId, delegate: delegateCodeStart)
+        
+        wait(for: [expectationCodeStart])
+        
+        // Correlation Id is validated internally against expectedContinueRequestParameters
+        delegateCodeStart.newState?.submitCode(code: "1234", delegate: delegateVerifyCode)
+        
+        wait(for: [expectationVerifyCode])
+        
+        // Correlation Id is validated internally against expectedTokenParams
+        delegateVerifyCode.newSignInAfterSignUpState?.signIn(scopes: ["scope1", "scope2"], delegate: delegateSignInAfterSignUp)
+        wait(for: [expectationSignInAfterSingUp])
+        
+        // User account result is validated internally against expectedUserAccountResult
+        XCTAssertTrue(delegateSignInAfterSignUp.onSignInCompletedCalled)
+    }
+    
+    // SignIn Password
+    // Testing SignInInitiate -> SignInChallenge -> SignInToken with Password
+    
+    func testSignInPassword_correlationId_whenSetOnStart_itCascadesToAll() {
+        let expectationPasswordStart = expectation(description: "Sign In Password Start")
+        let delegatePasswordStart = SignInPasswordStartDelegateSpy(expectation: expectationPasswordStart)
+        let expectationVerifyCode = expectation(description: "Sign In Verify Code")
+        let delegateVerifyCode = SignInVerifyCodeDelegateSpy(expectation: expectationVerifyCode)
+        
+        let signInRequestProviderMock = MSALNativeAuthSignInRequestProviderMock()
+        signInRequestProviderMock.mockInitiateRequestFunc(MSALNativeAuthHTTPRequestMock.prepareMockRequest())
+        signInRequestProviderMock.mockChallengeRequestFunc(MSALNativeAuthHTTPRequestMock.prepareMockRequest())
+        signInRequestProviderMock.expectedUsername = "username"
```

Comment:
You need to include `signInRequestProviderMock.expectedContext = contextMock`, otherwise it's not checking the correlationId

Reply:
Indeed, I had it and deleted it when I was cleaning up. Good catch.